### PR TITLE
Add replicated kvs example, add more lattice flow properties

### PR DESCRIPTION
**Pull request by MingweiSamuel**

### CODE REVIEW

File: hydroflow/examples/kvs_replicated/README.md
Comment (MingweiSamuel) [inline hydroflow/examples/kvs_replicated/README.md:1]: should update this

File: hydroflow_lang/src/graph/ops/cross_join.rs
Comment (MingweiSamuel) [inline hydroflow_lang/src/graph/ops/cross_join.rs:1]: add note on updated ops to commit messages

File: hydroflow/examples/kvs_replicated/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ - `GET <key>`
+ Commands are case-insensitive. All keys and values are treated as `String`s.
+
+## Running the example
+
+We will need four (or more) terminals:
+
+First, start a server:
+```
+cargo run -p hydroflow --example kvs_replicated -- --role "server" --addr localhost:12346
```
Comment (zzlk) [inline hydroflow/examples/kvs_replicated/README.md:21]: server is quoted but client is not, it should be consistent between the two documented example lines I think.

File: hydroflow/examples/kvs_replicated/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+```
+
+In another terminal connect a client to the first server:
```
Comment (zzlk) [inline hydroflow/examples/kvs_replicated/README.md:24]: Does this order have to be respected? Can you start up all the servers first?

File: hydroflow/examples/kvs_replicated/server.rs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+        // Join as a peer if peer_server is set.
+        source_iter_delta(peer_server) -> map(|peer_addr| (KvsMessage::PeerJoin, peer_addr)) -> network_send;
+
+        // join PUTs and GETs by key
+        writes -> map(|(key, value, _addr)| (key, value)) -> writes_store;
+        writes_store = persist() -> tee();
+        writes_store -> [0]lookup;
+        gets -> [1]lookup;
+        lookup = join();
```
Comment (zzlk) [inline hydroflow/examples/kvs_replicated/server.rs:42]: join_multiset?

what happens if you get two get requests in the same tick from the same client

File: hydroflow_lang/src/graph/ops/mod.rs [hunk 4]
```diff
@@ -258,24 +258,60 @@ pub const LATTICE_FOLD_REDUCE_FLOW_PROP_FN: FlowPropFn =
...
+                format!("`{}` expects lattice flow input, has sequential input. This may be an error in the future.", op_name),
+            )),
+        }
         Ok(vec![Some(FlowProps {
             star_ord: fp.new_star_ord(),
             lattice_flow_type: Some(LatticeFlowType::Cumul),
         })])
     };
 
+// TODO(mingwei):
```
Comment (zzlk) [inline hydroflow_lang/src/graph/ops/mod.rs:279]: what needs to be done

**Review by zzlk — APPROVED** [review]

File: hydroflow/examples/kvs_replicated/README.md (hunk 2 above)
Comment (MingweiSamuel) [reply hydroflow/examples/kvs_replicated/README.md:24]: yeah can be set up in any order as long as clients come after the servers they connect to

**Pull request merged**